### ISSUE TITLE
- Delete billing period: Created ticket to change to deactivate - low priority for now.

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#696]
In GitLab by @kdjstudios on Feb 19, 2018, 08:48

**Submitted by:** Kyle

**Helpdesk:** NA

**Server:** Internal/Merge

**Client/Site:** NA

**Account:** NA

**Issue:** 

We should update the delete billing period to deactivate/reactivate billing period.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 21, 2018, 05:27

Hi [Kyle] ( @kdjstudios ), 

Moving to next milestone with the same priority label.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 21, 2018, 05:27

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2019, 13:17

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2019, 13:17

unassigned @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 5, 2020, 23:39

@kdjstudios @jredfield 

Thank you for the update.

Please find below an estimate to activate/reactive functionality for the billing periods.

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Worked with Concepts, implementation ok|3|Low Impact|1.0|4|


* Do we need to send an email if any billing period is activated/reactivated?
  * If yes, we need to add `2` hrs more to develop the email functionality. and please provide the template, sender, recipients etc.. 


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 6, 2020, 12:51

@amishra @jredfield 

This seems extremely high, please elaborate on your description to further explain why it takes so long. Also why would it take 2 more hours for email functionality? We already have the functionality....

NOTE: Just like with all other deactivate/reactivate functionality it must send the emails to notify ops/site.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 7, 2020, 01:34

@kdjstudios @jredfield 

Thank you for the update.

Word Break:
* We will create routes for activate/deactivate. (Approx 15 min)
* We will update the status of activate/deactivate via Ajax.(Approx 2 hrs).
* Extra update like (creating a new branch commit push etc..)(Approx 20 min)
* Testing (Approx 1hrs)

> Also why would it take 2 more hours for email functionality? We already have the functionality....

Yes, we need to do create new template, methods etc..because notification email may be different from others.
 
Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 7, 2020, 01:34

assigned to @rranjan

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 7, 2020, 13:36

Yes, we need to do create new template, methods etc..because notification email may be different from others.
> Can you please calrify why we need new template, methods, etc. Copy/pasting the existing and adjusting them accordingly should take far less time. IE. Copying the above should take less then 10 minutes and then changing the "To:" address or other fields should take 20 seconds. Not 2 hours.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 8, 2020, 05:30

@kdjstudios @jredfield 

We have updated the estimate. we have added the time for the email notification in the estimate itself. 

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|3|Worked with Concepts, implementation ok|4|Low Impact|1.0|5|

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Jan 8, 2020, 06:16

@ranand We still would like an answer to this comment posted by @kdjstudios: http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/696#note_54135

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 8, 2020, 23:00

@kdjstudios 

> Can you please calrify why we need new template, methods, etc. Copy/pasting the existing and adjusting them accordingly should take far less time. IE. Copying the above should take less then 10 minutes and then changing the "To:" address or other fields should take 20 seconds. Not 2 hours.

I understand your concern, it was a bit high, so I updated it. Though it is already build functionality, we still need to follow the process to add it, which includes setup for this through backend or to check any references for it in the SAB or to do check for its basic functionality or format.

**We have a query that, do we need to send the email only when the billing period gets disabled or every time its status changes?**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 14, 2020, 10:46

We should be sending emails for both deactivation and reactivation, just like we have on Billing cycles, Activities, etc.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 24, 2020, 01:46

@kdjstudios @jredfield 

Thank you for the update.

We are using the following message at the time of billing period activation & deactivation. Please have a look let me know if we need to change.
**When we deactivate the billing period**

* Accounts are using that billing period under the accounts recurring fee then showing the alert message.

* The item [Billing Period Name] is currently in use by [6] accounts. You have been sent an email with a list of these accounts. To deactivate this #{field.name}. Please deactivate/remove the billing period from the account recurring fee for all accounts first. Email sent to: [site email address]

* Email Body:
Hello,

Please remove/deactivate the item [billing period name] Billing Period from the following accounts and then you will be able to deactivate the  Billing Period at the site level.

Accounts:
  * Demo1
  * Demo2

* No Billing Periods are selected on the account under the active account recurring fee section.
  * Body:  
    * User [current user name] with email address [email address] activate the Billing Period field from site [site name] at [current date]

**When reactivate the Billing Period:** 
*  Body: 
    * User [current user name] with email address [email address] reactivated Billing Period field from site [site name] at [current date].  
   Thank You!

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 24, 2020, 11:30

@amishra @ranand @jredfield 

Thank you please keep in mind the priority on this, as I see when have other higher priority items still waiting for estimates and development. 

But what is above looks good.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 10, 2020, 12:23

Updating priority.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Mar 16, 2020, 06:55

@kdjstudios  @jredfield 

Thanks for the update

We have completed the testing and setup branch on the merge2 server for your testing.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 16, 2020, 13:03

Looks good to me. marking for announcement.

EDIT: I just saw the email notice. Looks like we need a few adjustments.

1. Email address is incorrect.
2. Lets use past tense "deactivate" should be "deactivated"; and "reactivate" should be "reactivated". (No capital letters)
3. Remove the word "field" and also add a space after the Billing Period name.

![image](/uploads/ce0e084bbb16c42b9657bc80e7596cf0/image.png)

1. Use the following wording: 
> User [username] with email address [useremail] attempted to deactivate the Billing Period [name] from the site [sitename] at [date and time]. Please remove the Billing Period from the following accounts to deactivate at the site.
> 
> [list of account name and account number]
> 
> Thank you!

![image](/uploads/bad67a6c8d525387242c5366bd910add/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Mar 16, 2020, 23:26

@kdjstudios  @jredfield 

Thanks for the update.

> email address is incorrect.

Because its on development server so we had updated the email.

![image](/uploads/de1935e6d957699685fe22aa0f64fd50/image.png)

we have changed the email body as you have stated above.

![image](/uploads/a4ee1b8466305b55367c23ed03a8a0ec/image.png)
Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 17, 2020, 13:10

@rranjan 

Thank you for the updates. marking this for Announcement.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 30, 2020, 03:47

@kdjstudios @tim.traylor 

Thank you for the update.

We have successfully released on both of the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 30, 2020, 11:19

Looks good.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 30, 2020, 15:01

closed
